### DISCUSSION TITLE
Is there support for ignoring a local writer? [14684]

### DISCUSSION THREAD

**Discussion by MPW-waters** [eProsima/Fast-DDS#2703] [Support]
I have both a subscriber and publisher to the same topic. I want the subscriber to ignore the local publisher. I tried
DomainParticipant::ignore_publication(writer_->get_instance_handle()). It returns RETCODE_UNSUPPORTED. Does this mean that the ignore functionality is not implemented?

If ignore is not implemented, is there a way my subscriber can determine the writer for a given message and use that as a filter?

git log indicates I am using origin/2.6.x of Fast-DDS.

Thanks,
mw

**Comment by JLBuenoLopez** [comment]
Hi @MPW-waters,

You are correct. `DomainParticipant::ignore_publication` is not yet implemented as you can see in the [code](https://github.com/eProsima/Fast-DDS/blob/6a978a70589074f700d9b7c00cbaf91ba43d5031/src/cpp/fastdds/domain/DomainParticipant.cpp#L273). I see that [Fast DDS API reference](https://fast-dds.docs.eprosima.com/en/latest/fastdds/api_reference/dds_pim/domain/domainparticipant.html#_CPPv4N8eprosima7fastdds3dds17DomainParticipant18ignore_publicationERK16InstanceHandle_t) is not showing the warning indicating this, so I will prepare a PR to include a warning in our doxygen documentation so it is clear which API is implemented and which is not available to the user as of now.

If the subscriber and publisher are run in different process you can use the [ignore Participant flags](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/general_disc_settings.html#ignore-participant-flags). If this is not the case, then you can either redesign your application to have two different Participants, one dedicated to publication and the other one to subscription; or you can check the writer GUID when you take the local sample and dropped it if it is coming from the local publisher when you are reading/taking it (this is not really efficient as the sample is in fact being sent and you are just dropping it when your application is notified).

Finally, in case that you are interested on accelerating the development of this API and other unsupported APIs you might need, you can contact our [support team](https://github.com/eProsima/Fast-DDS#getting-help) for more information.

**Comment by MPW-waters** [comment]
ignore Participant flags worked great. Thank you for your quick and helpful response.
